engine/schema/src/org/apache/cloudstack/outofbandmanagement/dao/OutOfBandManagementDaoImpl.java

```diff
@@ -0,0 +1,163 @@
+            PreparedStatement pstmt = txn.prepareAutoCloseStatement(sql);
+            pstmt.setLong(1, resource);
+            pstmt.executeUpdate();
+            txn.commit();
+        } catch (SQLException e) {
+            txn.rollback();
+            throw new CloudRuntimeException("Unable to reset out-of-band management ownership based on resource:" + resource);
+        }
+    }
+
+    @Override
+    public void expireOutOfBandManagementOwnershipByHours(long hours) {
+        final String resetOwnerSql = "UPDATE oobm set mgmt_server_id=NULL where update_time<= (NOW() - INTERVAL ? HOUR)";
+        executeExpireOwnershipSql(resetOwnerSql, hours);
+    }
+
+    @Override
+    public void expireOutOfBandManagementOwnershipByServer(long serverId) {
+        final String resetOwnerSql = "UPDATE oobm set mgmt_server_id=NULL, power_state=NULL where mgmt_server_id=?";
+        executeExpireOwnershipSql(resetOwnerSql, serverId);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Expired out-of-band management ownership for hosts owned by management server id:" + serverId);
+        }
+    }
+
+    @Override
+    public boolean updateState(OutOfBandManagement.PowerState oldStatus, OutOfBandManagement.PowerState.Event event, OutOfBandManagement.PowerState newStatus, OutOfBandManagement vo, Object data) {
+        // lock target row from beginning to avoid lock-promotion caused deadlock
+        OutOfBandManagementVO oobmHost = lockRow(vo.getId(), true);
+        if (oobmHost == null) {
+            if (LOG.isTraceEnabled()) {
+                LOG.trace("Failed to lock row to update state for out-of-band management host id: " + vo.getHostId());
+            }
+            return false;
+        }
+
+        Long newManagementServerId = event.getServerId();
+        // Avoid updates when old ownership and state are same as new
+        if (oldStatus == newStatus && (oobmHost.getManagementServerId() != null && oobmHost.getManagementServerId().equals(newManagementServerId))) {
+            return false;
+        }
+
+        if (event == OutOfBandManagement.PowerState.Event.Disabled) {
+            newManagementServerId = null;
+        }
+
+        SearchCriteria<OutOfBandManagementVO> sc = StateUpdateSearch.create();
+        sc.setParameters("status", oldStatus);
+        sc.setParameters("id", oobmHost.getId());
+        sc.setParameters("update", oobmHost.getUpdateCount());
+
+        oobmHost.incrUpdateCount();
+        UpdateBuilder ub = getUpdateBuilder(oobmHost);
+        ub.set(oobmHost, PowerStateAttr, newStatus);
+        ub.set(oobmHost, UpdateTimeAttr, DateUtil.currentGMTTime());
+        ub.set(oobmHost, MsIdAttr, newManagementServerId);
+
+        int result = update(ub, sc, null);
+        if (LOG.isDebugEnabled() && result <= 0) {
+            LOG.debug(String.format("Failed to update out-of-band management power state from:%s to:%s due to event:%s for the host id:%d", oldStatus, newStatus, event, oobmHost.getHostId()));
```

Comment:
It is triggered by background sync thread, I've avoided to use WARN to spam the logs file in case of large prod. In case of frequent updates this can cause a lot of log entries, and when states are same.